### PR TITLE
Modernized rule PHPStan\Rules\DeadCode\NoopRule

### PR DESCRIPTION
**Pull request by sspat**
Issue: https://github.com/phpstan/phpstan/issues/2582

### CODE REVIEW

**Comment by ondrejmirtes** [comment]
Yes, perfect! :)

File: tests/PHPStan/Rules/DeadCode/NoopRuleTest.php [hunk 1]
```diff
@@ -6,6 +6,9 @@
 use PHPStan\Rules\Rule;
 use PHPStan\Testing\RuleTestCase;
 
+/**
+ * @extends \PHPStan\Testing\RuleTestCase<NoopRule>
```
Comment (lookyman) [inline tests/PHPStan/Rules/DeadCode/NoopRuleTest.php]: Should this not be FQN rule class?

Comment (sspat) [reply tests/PHPStan/Rules/DeadCode/NoopRuleTest.php]: It is, the test and the rule class are in the same namespace.

Comment (lookyman) [reply tests/PHPStan/Rules/DeadCode/NoopRuleTest.php]: Right, of course, by bad. 👍

**Pull request merged**